presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -209,36 +176,57 @@ public static Slice stPoint(@SqlType(DOUBLE) double x, @SqlType(DOUBLE) double y
     @SqlType(GEOMETRY_TYPE_NAME)
     public static Slice stMultiPoint(@SqlType("array(" + GEOMETRY_TYPE_NAME + ")") Block input)
     {
-        MultiPoint multipoint = new MultiPoint();
+        CoordinateSequence coordinates = readPointCoordinates(input, "ST_MultiPoint", false);
+        if (coordinates.size() == 0) {
+            return null;
+        }
+
+        return serialize(createJtsMultiPoint(coordinates));
+    }
+
+    private static CoordinateSequence readPointCoordinates(Block input, String functionName, boolean forbidDuplicates)
+    {
+        PackedCoordinateSequenceFactory coordinateSequenceFactory = new PackedCoordinateSequenceFactory();
+        double[] coordinates = new double[2 * input.getPositionCount()];
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).